apps/labs/posts/pep-517-build-system-popularity.md

```diff
@@ -0,0 +1,615 @@
+script from the CPython repository. The downloaded files corresponded
+to the newest versions available on 2025-01-08. Then I unpacked `pyproject.toml`,
+`setup.cfg` and `setup.py` files from them.
+
+Out of 8000 projects listed in the dump:
+
+- two were not available anymore,
+
+- 561 did not feature source distributions at all,
+
+- two did not have any build system files,
+
+- one had incorrectly capitalized filenames, preventing it from working
+  on Linux.
+
+I performed the analysis on the remaining 7434 source distributions.
+However, I must note that not all of these distributions could actually
+be installed, as I explain in detail further on.
+
+I wrote a few scripts to analyze these distributions and process
+the results, creating tables that are included in this post. The scripts
+are available in the [pep517-stats](https://github.com/mgorny/pep517-stats/)
+repository. They perform the following actions:
+
+1. Obtain the raw values of `build-system.build-backend` key from `pyproject.toml`
+   to determine the popularity of individual build backends.
+
+2. Match these values to known build systems, combining multiple backends
+   corresponding to the same or closely related packages.
+
+3. Look at the `build-system.requires` key in packages using a custom build
+   backend to determine which package it is based on.
+
+4. Determine which of the different configuration formats supported
+   by setuptools are used by the project:
+
+   - `pyproject.toml` — if `[project]` table exists in said file
+   - `setup.cfg` — if `[metadata]` section exists in said file
+   - `setup.py` — if said file exists
+
+5. Run the `get_requires_for_build_wheel()` hook
+   to obtain all build requirements of the package.
+
+## The most popular build systems
+
+<div style={{textAlign: 'center'}}>
+
+<table id="table-1" style={{width: 'auto', margin: '0 2em', display: 'inline-block', verticalAlign: 'top'}}>
+  <caption>Table 1. Cumulative backend use counts</caption>
+  <tr><th>Family or backend</th><th>Count</th></tr>
+  <tr style={{ background: '#fef' }}><td style={{ height: '4em' }}>setuptools</td><td align='right'>5854</td></tr>
+  <tr style={{ background: '#eff' }}><td style={{ height: '4em' }}>poetry</td><td align='right'>625</td></tr>
+  <tr style={{ background: '#ffe' }}><td style={{ height: '4em' }}>hatchling</td><td align='right'>480</td></tr>
+  <tr style={{ background: '#eef' }}><td style={{ height: '4em' }}>flit</td><td align='right'>285</td></tr>
+  <tr><td style={{ height: '3em' }}>`maturin`</td><td align='right'>85</td></tr>
+  <tr style={{ background: '#efe' }}><td style={{ height: '4em' }}>pdm</td><td align='right'>42</td></tr>
+  <tr style={{ background: '#fee' }}><td style={{ height: '4em' }}>scikit-build-core</td><td align='right'>30</td></tr>
+  <tr><td style={{ height: '3em' }}>`mesonpy`</td><td align='right'>16</td></tr>
```

Comment:
minor: could you rename to `meson-python`

Reply:
I can, though I'm using the backend Python package name here, which is `mesonpy` (it's taken straight from `pyproject.toml`).

Reply:
The Python package name is `meson-python`; the import name is `mesonpy`. You're also using `scikit-build-core` for example (with dashes), which isn't the import name but the package name. There's a bit of inconsistency here. I think package names are clearer.